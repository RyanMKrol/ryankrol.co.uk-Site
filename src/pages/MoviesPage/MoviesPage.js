import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import styles from './MoviesPage.module.css';

import { DataCollection, MoviesItem } from '../../components';
import { baseEndpoint } from '../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class MoviesPage extends Component {
  /**
   * @param {object} props object containing a single prop - fullSize,
   *                       which dictates how much data to display
   * @class
   */
  constructor(props) {
    super();

    this.endpoint = `${baseEndpoint()}/movies`;
    this.outputLimit = props.fullSize ? undefined : 20;
    this.readMore = props.fullSize ? null : (
      <Link className={styles.seeMore} to={'/movies'}>
        <h1 className="highlighted">...</h1>
      </Link>
    );
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h1 className="light">My Films</h1> <br />
        <p>Literally just a bunch of movies I own, idk</p>
        <DataCollection
          outputSize={this.outputLimit}
          endpoint={this.endpoint}
          itemTag={MoviesItem}
        />
        {this.readMore}
      </div>
    );
  }
}

export default MoviesPage;
