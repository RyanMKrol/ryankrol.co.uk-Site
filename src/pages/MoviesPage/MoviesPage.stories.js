import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';

import { FULL_SIZE_MOCK, LIMITED_SIZE_MOCK } from './__mocks__/MoviesPageMocks';

export default {
  title: 'Pages/MoviesPage',
};

export const FullSize = () => <Router>{FULL_SIZE_MOCK}</Router>;
export const LimitedSize = () => <Router>{LIMITED_SIZE_MOCK}</Router>;
