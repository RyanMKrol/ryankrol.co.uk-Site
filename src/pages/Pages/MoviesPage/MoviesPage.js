import React, { Component } from 'react';

import styles from './MoviesPage.module.css';

import { DataCollection, MoviesItem } from '../../../components';
import { baseEndpoint } from '../../../utils/ProdUtils';

/**
 * Page for rendering sub-section of Ratings content
 */
class MoviesPage extends Component {
  /**
   * @class
   */
  constructor() {
    super();

    this.endpoint = `${baseEndpoint()}/movies`;
  }

  /**
   * Render method
   *
   * @returns {string} JSX Content
   */
  render() {
    return (
      <div className={styles.container}>
        <h2>Literally just a bunch of movies I own, idk</h2>
        <br />
        <DataCollection endpoint={this.endpoint} itemTag={MoviesItem} />
      </div>
    );
  }
}

export default MoviesPage;
