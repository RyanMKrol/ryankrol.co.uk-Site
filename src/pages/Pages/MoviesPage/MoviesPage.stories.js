import React from 'react';

import MoviesPage from './MoviesPage';

export default {
  title: 'Pages/MoviesPage',
};

/**
 * default story
 *
 * @returns {React.Component} JSX for the MoviesPage
 */
export const normal = () => <MoviesPage />;
