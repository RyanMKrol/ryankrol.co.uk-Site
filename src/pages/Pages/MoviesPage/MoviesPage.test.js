import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import MoviesPage from './MoviesPage';

it('renders MoviesPage correctly', () => {
  const header = shallow(<MoviesPage />);
  expect(toJson(header)).toMatchSnapshot();
});
