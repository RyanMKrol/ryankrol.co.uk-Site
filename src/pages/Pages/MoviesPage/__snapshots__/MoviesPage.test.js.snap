// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MoviesPage correctly 1`] = `
<div
  className="container"
>
  <h2>
    Literally just a bunch of movies I own, idk
  </h2>
  <br />
  <DataCollection
    endpoint="http://www.ryankrol.co.uk/api/movies"
    itemTag={[Function]}
  />
</div>
`;
